config/user-libatomic.m4

```diff
@@ -0,0 +1,20 @@
+dnl #
+dnl # If -latomic exists, it's needed for __atomic intrinsics.
```

Comment:
Should setting `-latomic` be mandatory now?  i.e. does it make sense for the configure to outright fail here if the flag isn't supported?  Or are there architectures where's it's really just not needed at all.

Reply:
FreeBSD 13 doesn't have libatomic (at least on amd64, but I think it's a clang thing), and therefore can't possibly need it; linking to libatomic fails on some of the ancienter CentOSes on the buildds, but just not linking to instead works, hence this selection logic

Reply:
It be nice to include this additional information in the comment here.  Specifically, it's not needed on FreeBSD 13 and older CentOS's presumably because the much older version of gcc?

Reply:
Dunno, I've honestly given up attempting to comprehend Why CentOS Is Broken In Another Insane Way Today.

Added a few paragraphs below this line.